cmd/clusterctl/client/yamlprocessor/simple_processor_test.go

```diff
@@ -80,7 +95,12 @@ func TestSimpleProcessor_GetVariables(t *testing.T) {
 			args: args{
 				data: "yaml with ${C:=default}\n${B}\n${A=foobar}",
 			},
-			want: []string{"A", "B", "C"},
+			// want: []VariableMetadata{"A", "B", "C"},
+			want: []VariableMetadata{
+				&variableMetadata{name: "A", required: false, defaultValue: "foobar"},
+				&variableMetadata{name: "B", required: true, defaultValue: ""},
+				&variableMetadata{name: "C", required: false, defaultValue: "default"},
+			},
```

Comment:
@mabikash I think we can simply improve test by creating two additional test cases similar to "variables with envsubst functions are properly parsed"; the new test should focus explixitly on checking the properties we are extracting in VariablesMetadata:
- "default values are properly parsed" (and test some edge case like string, numbers, empty string)
- "required are properly parsed" (and test required true/false)

Reply:
actually default value is of type string 
type variableMetadata struct {
	name         string
	required     bool
	defaultValue string
}
so all dafault value are string not number.

Reply:
everything should be fixed

Reply:
@fabriziopandini all test complete